logging/zap/grpclogger.go

```diff
@@ -44,3 +46,103 @@ func (l *zapGrpcLogger) Printf(format string, args ...interface{}) {
 func (l *zapGrpcLogger) Println(args ...interface{}) {
 	l.logger.Info(fmt.Sprint(args...))
 }
+
+// ReplaceGrpcLoggerV2 replaces the grpc_log.LoggerV2 with the input zap.Logger
+// This logger adheres to the grpc go environment variables GRPC_GO_LOG_VERBOSITY_LEVEL and GRPC_GO_LOG_SEVERITY_LEVEL.
+func ReplaceGrpcLoggerV2(logger *zap.Logger) {
+	zgl := &zapGrpcLoggerV2{}
+	verbosity := os.Getenv("GRPC_GO_LOG_VERBOSITY_LEVEL")
+	if v, err := strconv.Atoi(verbosity); err == nil {
+		zgl.verbosity = v
+	}
+
+	logLevel := os.Getenv("GRPC_GO_LOG_SEVERITY_LEVEL")
+	switch logLevel {
+	case "", "ERROR", "error": // If env is unset, set level to ERROR.
+		zgl.severity = errorLevel
+	case "WARNING", "warning":
+		zgl.severity = warnLevel
+	case "INFO", "info":
+		zgl.severity = infoLevel
+	}
+
+	zgl.Logger = logger.With(zap.String("system", "grpc"), zap.Bool("grpc_log", true))
+	grpclog.SetLoggerV2(zgl)
+
+}
+
+const (
+	errorLevel = iota
+	warnLevel
+	infoLevel
+)
+
+type zapGrpcLoggerV2 struct {
+	*zap.Logger
+	verbosity int
+	severity  int
+}
+
+func (l *zapGrpcLoggerV2) Info(args ...interface{}) {
+	if l.severity >= infoLevel {
+		l.Info(fmt.Sprint(args...))
+	}
+}
+
+func (l *zapGrpcLoggerV2) Infoln(args ...interface{}) {
+	if l.severity >= infoLevel {
+		l.Info(fmt.Sprint(args...))
+	}
+}
+
+func (l *zapGrpcLoggerV2) Infof(format string, args ...interface{}) {
+	if l.severity >= infoLevel {
+		l.Info(fmt.Sprintf(format, args...))
+	}
+}
+
+func (l *zapGrpcLoggerV2) Warning(args ...interface{}) {
+	if l.severity >= warnLevel {
+		l.Warn(fmt.Sprint(args...))
+	}
+}
+
+func (l *zapGrpcLoggerV2) Warningln(args ...interface{}) {
+	if l.severity >= warnLevel {
+		l.Warn(fmt.Sprint(args...))
+	}
+}
+
+func (l *zapGrpcLoggerV2) Warningf(format string, args ...interface{}) {
+	if l.severity >= warnLevel {
+		l.Warn(fmt.Sprintf(format, args...))
+	}
+}
+
+func (l *zapGrpcLoggerV2) Error(args ...interface{}) {
+	l.Error(fmt.Sprint(args...))
```

Comment:
does this not need to be `l.logger.Error`?

Reply:
same comment on the rest of the funcs

Reply:
it doesn't need to because the logger is promoted.

``` go
type zapGrpcLoggerV2 struct {
	*zap.Logger
	verbosity int
	severity  int
}
```

Reply:
@domgreen is right,  since the embedded type's method is the same name, see this [quick stack](https://stackoverflow.com/questions/20861786/golang-struct-calling-embedded-type-methods-when-method-has-been-overloaded) for another example.

Heres a plug for using GoLand, as it shows how this method is recursive, which is another way of finding out the zap Logger is not being called.
![image](https://user-images.githubusercontent.com/877194/50879368-87234f00-138f-11e9-9865-0954c02fe61a.png)